proposals/0197-chilli-peppers.md

```diff
@@ -0,0 +1,147 @@
+All accounts will have a Chili Pepper Clock Timestamp, which will only exist for accounts which are hot, and is discarded for accounts which are cold. All implementations should keep track of the current hot accounts.
+ 
+#### Hot and Cold Account Designation
+ 
+An account is designated as cold when its Account Chili Pepper Clock Timestamp falls behind the current Block Chili Pepper Clock by more than the Hot Cache Size parameter.
+An account which has never existed is considered cold. An account that is deleted is still considered hot until its state unit clock lapses into cold. Creating an account against a
+deleted account which is still hot, will create the hot account again.
+ 
+#### Storage and Management
+ 
+To manage the Account State Unit Clocks efficiently, Solana employs a table, associating each hot account with its respective Chili Pepper Clock Timestamp. This table enables the dynamic tracking and updating of account states, facilitating the transition of accounts between hot and cold statuses based on their activity.
+ 
+### Error Cases for State Units Implementation
+ 
+Here are common error scenarios related to State Units and their respective handling mechanisms:
+ 
+#### Exceeding Block Chili Pepper Limit
+ 
+- **Error Description**: This error occurs when a transaction's requested Chili Peppers exceed the remaining capacity of the Block Chili Pepper Clock for the current block.
+- **Handling**: The block is marked as invalid and cannot be processed.
+ 
+#### Invalid Chili Pepper Request
+ 
+- **Error Description**: A transaction specifies an invalid number of Chili Peppers, either by requesting more than a predefined maximum limit per transaction or by formatting the request improperly.
+- **Handling**: The transaction is invalidated, and an "Invalid Chili Pepper Request" error message is issued. Developers must ensure that Chili Pepper requests conform to protocol specifications, including proper formatting and adherence to maximum limits.
+ 
+#### Account State Unit Clock Synchronization Failure
+ 
+#### Accessing Cold Account
+ 
+- **Error Description**: A transaction attempts to interact with an account that has been designated as "cold" due to its Chili Pepper Clock Timestamp falling below the "Hot Cache Size" threshold relative to the current Block Chili Pepper Clock, without requesting sufficient Chili Peppers.
+- **Handling**: The transaction is rejected with a "Cold Account Access Attempted" error.
+ 
+ 
+## Alternatives Considered
+ 
+What alternative designs were considered and what pros/cons does this feature
+have relative to them?
+ 
+## Impact
+ 
+How will the implemented proposal impacts dapp developers, validators, and core contributors?
+ 
+## Security Considerations
+ 
+What security implications/considerations come with implementing this feature?
```

Comment:
chili peppers are a consumable resource without cost?   Does this mean I can 

1) put a single txn with all the chili peppers possible on a block 
2) a high enough priority fee to guarantee it gets put into the block 

and lock out everybody elses ability to retrieve cold accounts?

I would guess we would limit the max chili peppers a single txn could request to be the max number of accounts it could reference times 10m...  Given ALUT, that would be a lot of chili peppers.. all of them

I would suggest we make chili peppers be representative of actual SOL so that it would cost to do this attack where we only actually charge for actual cold access?!  a real head scratcher...

Reply:
A block builder sees a txn, notices it references unknown accounts, notices it does not have the chili peppers required to account for the meta data for those unknown accounts, does not include that txn

I am now thinking hitting cold meta data needs to be super expensive..